Makefile

```diff
@@ -123,15 +127,30 @@ ssj_sources=src/ssj/main.c \
    src/ssj/parser.c \
    src/ssj/session.c
 
+ifeq ($(os), Darwin)
+LINKER_ARGS=-Wl,-rpath,\$$ORIGIN
+engine_libs+=-lallegro_main
```

Comment:
Note that while `allegro_main` is only specifically *required* on macOS, it's safe to link to it on Linux also (a dummy lib is provided). Up to you whether you want to make that change to simplify things or not.

Reply:
Yeah I do think that's better. Who knows, it may be needed on future platforms.